### PR TITLE
180 - Feature/update remix version

### PR DESCRIPTION
**Pull request by juanan150**
# What does this PR do?

This PR updates the Remix version to the version `1.18.0`

# Where should the reviewer start?

Run `npm i` (ideally to delete the node_modules before running it).

# How should this be manually tested?

Run `npm i` and test that the functionalities are ok

# What are the relevant tickets?

#180 

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [X] I added tests to prove that my fix is effective or my feature works.
- [X] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

File: app/utils.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
```
Comment (jackbravo) [inline app/utils.ts:1]: Why is this needed? I removed it locally and got no errors.

File: app/utils.ts [hunk 2]
```diff
@@ -87,3 +89,13 @@ export const generateRandomNumberString = () => {
 
   return randomString;
 };
+
+export const validateNavigationRedirect = (navigation: Navigation) => {
+  return (
+    navigation.state === "loading" &&
+    navigation.formMethod !== null &&
+    navigation.formMethod !== "GET" // &&
```
Comment (jackbravo) [inline app/utils.ts:97]: Locally I was getting this type error: `This comparison appears to be unintentional because the types 'FormMethod | undefined' and '"GET"' have no overlap.ts(2367)`

And indeed, when inspecting FormMethod, I see:

```typescript
export declare type MutationFormMethod = "post" | "put" | "patch" | "delete";
export declare type FormMethod = "get" | MutationFormMethod;
```

So changing to this fixes the error:

```suggestion
    navigation.formMethod !== "get" // &&
```

**Review by jackbravo — APPROVED** [review]
🥇 💯

**Comment by jackbravo** [comment]
Also, just as a comment, I got this message when running `npm install` 😊 👍🏼 

```
added 93 packages, removed 203 packages, changed 133 packages, and audited 1597 packages in 7s

12 vulnerabilities (5 moderate, 1 high, 6 critical)

To address issues that do not require attention, run:
  npm audit fix
```

**Comment by jackbravo** [comment]
By the way, when doing `npm run dev` I got the following warning about future v2 deprecations that we can address already while in v1. Probably we should open tickets for them?

What do you think?

```
 warn  The `formMethod` API is changing in v2
┃ You can use the `v2_normalizeFormMethod` future flag to opt-in early.
┃ -> https://remix.run/docs/en/v1.15.0/pages/v2#formMethod
┗
 warn  The route `meta` API is changing in v2
┃ You can use the `v2_meta` future flag to opt-in early.
┃ -> https://remix.run/docs/en/v1.15.0/pages/v2#meta
┗
 warn  The route `headers` API is changing in v2
┃ You can use the `v2_headers` future flag to opt-in early.
┃ -> https://remix.run/docs/en/v1.17.0/pages/v2#route-headers
┗
 warn  The default server module format is changing in v2
┃ The default format will change from `cjs` to `esm`.
┃ You can keep using `cjs` by explicitly specifying `serverModuleFormat: 'cjs'`.
┃ You can opt-in early to this change by explicitly specifying `serverModuleFormat: 'esm'`
┃ -> https://remix.run/docs/en/v1.16.0/pages/v2#servermoduleformat
┗
 warn  The route file convention is changing in v2
┃ You can use the `v2_routeConvention` future flag to opt-in early.
┃ -> https://remix.run/docs/en/v1.15.0/pages/v2#file-system-route-convention
┗
```

File: app/utils.ts (hunk 1 above)
Comment (jackbravo) [reply app/utils.ts:1]: nevermind, I see now it is the

```typescript
  const crypto = require("crypto");
```

in the ``generateRandomNumberString` function

Comment (jackbravo) [reply app/utils.ts:1]: But maybe instead we should disable it for just that line instead of the entire file.

File: app/utils.ts (hunk 2 above)
Comment (jackbravo) [reply app/utils.ts:97]: Adding `v2_normalizeFormMethod: true,` to `remix.config.js` also fixes this error. I like that approach more.

**Comment by jackbravo** [comment]
Did the easy ones :-p. No only these remain:

 warn  The route `meta` API is changing in v2
┃ You can use the `v2_meta` future flag to opt-in early.
┃ -> https://remix.run/docs/en/v1.15.0/pages/v2#meta
┗
 warn  The default server module format is changing in v2
┃ The default format will change from `cjs` to `esm`.
┃ You can keep using `cjs` by explicitly specifying `serverModuleFormat: 'cjs'`.
┃ You can opt-in early to this change by explicitly specifying `serverModuleFormat: 'esm'`
┃ -> https://remix.run/docs/en/v1.16.0/pages/v2#servermoduleformat
┗
 warn  The route file convention is changing in v2
┃ You can use the `v2_routeConvention` future flag to opt-in early.
┃ -> https://remix.run/docs/en/v1.15.0/pages/v2#file-system-route-convention
┗

**Comment by juanan150** [comment]
Wow! I just read all this. Thanks a lot @jackbravo! :+1:
